modules/dxp/apps/portal-search-tuning/portal-search-tuning-rankings-web-test/src/testIntegration/java/com/liferay/portal/search/tuning/rankings/web/internal/upgrade/BaseRankingUpgradeProcessTestCase.java

```diff
@@ -0,0 +1,155 @@
+/**
+ * SPDX-FileCopyrightText: (c) 2024 Liferay, Inc. https://liferay.com
+ * SPDX-License-Identifier: LGPL-2.1-or-later OR LicenseRef-Liferay-DXP-EULA-2.0.0-2023-06
+ */
+
+package com.liferay.portal.search.tuning.rankings.web.internal.upgrade;
+
+import com.liferay.change.tracking.test.util.BaseCTUpgradeProcessTestCase;
+import com.liferay.counter.kernel.service.CounterLocalService;
+import com.liferay.json.storage.model.JSONStorageEntry;
+import com.liferay.json.storage.service.JSONStorageEntryLocalService;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONUtil;
+import com.liferay.portal.kernel.model.change.tracking.CTModel;
+import com.liferay.portal.kernel.service.ClassNameLocalService;
+import com.liferay.portal.kernel.service.change.tracking.CTService;
+import com.liferay.portal.kernel.test.util.TestPropsValues;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.search.tuning.rankings.index.Ranking;
+import com.liferay.portal.search.tuning.rankings.index.name.RankingIndexName;
+import com.liferay.portal.search.tuning.rankings.index.name.RankingIndexNameBuilder;
+import com.liferay.portal.test.rule.Inject;
+import com.liferay.portal.upgrade.registry.UpgradeStepRegistrator;
+import com.liferay.portal.upgrade.test.util.UpgradeTestUtil;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+
+import org.junit.After;
+import org.junit.BeforeClass;
+
+/**
+ * @author Almir Ferreira
+ */
+public abstract class BaseRankingUpgradeProcessTestCase
+	extends BaseCTUpgradeProcessTestCase {
```

Comment:
Is BaseCTUpgradeProcessTestCase necessary in both upgrades? if not please use a Util to avoid duplicated code instead of abstract class.

Reply:
Hi @gustavolimav, it is important to ensure that the CtCollectionId is not lost after the upgrade. It's interesting to test for both upgrade processes